patches/@sentry+react-native+1.7.1.patch

```diff
@@ -0,0 +1,205 @@
++            collection.forEach(function (item, index) {
++                SyncPromise.resolve(item)
++                    .then(function (value) {
++                    resolvedCollection[index] = value;
++                    counter -= 1;
++                    if (counter !== 0) {
++                        return;
++                    }
++                    resolve(resolvedCollection);
++                })
++                    .then(null, reject);
++            });
++        });
++    };
++    /** JSDoc */
++    SyncPromise.prototype.then = function (onfulfilled, onrejected) {
++        var _this = this;
++        return new SyncPromise(function (resolve, reject) {
++            _this._attachHandler({
++                onfulfilled: function (result) {
++                    if (!onfulfilled) {
++                        // TODO: ¯\_(ツ)_/¯
++                        // TODO: FIXME
++                        resolve(result);
++                        return;
++                    }
++                    try {
++                        resolve(onfulfilled(result));
++                        return;
++                    }
++                    catch (e) {
++                        reject(e);
++                        return;
++                    }
++                },
++                onrejected: function (reason) {
++                    if (!onrejected) {
++                        reject(reason);
++                        return;
++                    }
++                    try {
++                        resolve(onrejected(reason));
++                        return;
++                    }
++                    catch (e) {
++                        reject(e);
++                        return;
++                    }
++                },
++            });
++        });
++    };
++    /** JSDoc */
++    SyncPromise.prototype.catch = function (onrejected) {
++        return this.then(function (val) { return val; }, onrejected);
++    };
++    /** JSDoc */
++    SyncPromise.prototype.finally = function (onfinally) {
++        var _this = this;
++        return new SyncPromise(function (resolve, reject) {
++            var val;
++            var isRejected;
++            return _this.then(function (value) {
++                isRejected = false;
++                val = value;
++                if (onfinally) {
++                    onfinally();
++                }
++            }, function (reason) {
++                isRejected = true;
++                val = reason;
++                if (onfinally) {
++                    onfinally();
++                }
++            }).then(function () {
++                if (isRejected) {
++                    reject(val);
++                    return;
++                }
++                // tslint:disable-next-line:no-unsafe-any
++                resolve(val);
++            });
++        });
++    };
++    return SyncPromise;
++  }());
++
++export function flush(timeout) {
```

Comment:
is all that just for flush() support ? we could just to adjust our code to make sure we copied error message for show it them on dialog/ do other stuff before logging (like we did in goodserver). this allows us to keep sync interface for log.error and don't use mutate-restore on Sentry.flush().finally()

Reply:
Yes, this implementation it’s all for the flush issue. What do you refer when saying good-server implementation? Could you explain more about this, please?